src/lib.rs

```diff
@@ -276,7 +279,24 @@ pub fn download_bill_text(bill_url: &Url) {
     // Select, and print each paragraph of the bill text
     let text_selector = Selector::parse("div.modal-body div").unwrap();
     let text_element = text_document.select(&text_selector).next().unwrap();
-    for node in text_element.text().collect::<Vec<_>>() {
-        println!("{node}");
+    let text_nodes = text_element.text().collect::<Vec<_>>();
+    for text_node in &text_nodes {
+        println!("{text_node}");
+    }
+
+    // Print each paragraph of the bill text to a file, if specified
+    if let Some(output_filename) = output_option {
```

Comment:
Would it be better to separate getting and printing? Use one function to get the text_nodes and return them, then call a section function in main to decide how to handle printing/writing, etc? It could help reduce coupling if we want to change how things are outputted

Reply:
I agree with this ... however ... I could not solve the ownership problem. Perhaps an exercise to try together?